pandas/_libs/missing.pyx

```diff
@@ -160,7 +161,11 @@ def isnaobj_old(arr: ndarray) -> ndarray:
     result = np.zeros(n, dtype=np.uint8)
     for i in range(n):
         val = arr[i]
-        result[i] = val is NaT or _check_none_nan_inf_neginf(val)
+        result[i] = (
```

Comment:
is this the sam construct we have in isnaobj?

Reply:
We use checknull there, I'll try to incorporate that here